# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsGridContainerFrame.cpp
# Commit: f12c5ebd0bfc
# Full Hash: f12c5ebd0bfce0915e7979bc95e7e8781c2f3a4f
# Author: Mats Palmgren <mats@mozilla.com>
# Date: 2019-05-24 21:49:59
# Description:
#   Bug 1553824 - [css-grid-2] Invalidate any stored used track sizes on subgrids too.  r=dholbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D32372
# ==============================================================================

diff -r 386097a10f84 -r f12c5ebd0bfc layout/generic/nsGridContainerFrame.cpp
--- a/layout/generic/nsGridContainerFrame.cpp	Thu May 23 19:03:36 2019 +0300
+++ b/layout/generic/nsGridContainerFrame.cpp	Thu May 23 21:11:48 2019 +0200
@@ -3857,7 +3857,7 @@
 
   mAreas = aState.mFrame->GetImplicitNamedAreas();
 
-  if (aState.mFrame->HasSubgridItems()) {
+  if (aState.mFrame->HasSubgridItems() || aState.mFrame->IsSubgrid()) {
     if (auto* uts = aState.mFrame->GetUsedTrackSizes()) {
       uts->mCanResolveLineRangeSize = { false, false };
       uts->mSizes[eLogicalAxisInline].ClearAndRetainStorage();
